cpp/oneapi/dal/backend/primitives/selection/test/kselect_by_rows_dpc.cpp

```diff
@@ -69,9 +69,12 @@ class selection_by_rows_test : public te::policy_fixture {
     void check_results(const ndview<float_t, 2>& data,
                        const ndview<float_t, 2>& selection,
                        const ndview<int, 2>& indices) {
-        ONEDAL_ASSERT(data.get_dimension(0) == selection.get_dimension(0));
-        ONEDAL_ASSERT(data.get_dimension(0) == indices.get_dimension(0));
-
+        if (selection_out) {
+            ONEDAL_ASSERT(data.get_dimension(0) == selection.get_dimension(0));
```

Comment:
Isn't it better not to introduce `if` branch just for assertion but insert the condition in the assert statement itself? 

```c++
ONEDAL_ASSERT(selection_out && data.get_dimension(0) == selection.get_dimension(0));
```